### DISCUSSION TITLE
Improve usability of effect parameters

### DISCUSSION THREAD

**Discussion by VanCastar** [4ian/GDevelop#3504] [Enhancement ideas (and their technical discussions)]
The parameter page of effects is a bit unintuitive and clumsy to use and thus the process is prone to annoying mistakes.
Too many extra steps to just get parameter names visible and having to remember and type them correctly over in the scene editor needing to go back and forth just to check what the parameter designations were.

Suggestions to improve usability and lessening of error probabilities.

1) Make effect parameter names a pull-down list of strings (when editing an action that changes an effect parameter)
2) Display the parameter "real" name after the parameter "friendly" name :  CenterX (Parameter: "centerX")
3) Add a long description to effect parameters that will be displayed under the input field (this would match the UI pattern used for function parameters and behavior properties)

**Comment by tristanbob** [comment]
For added clarity, item 1 involves changes to the event sheet editor for all conditions/actions/expressions related to effects (on objects and layers).  

**Bonus points** will be awarded if the "Effect Name" also provides a pull-down list of all configured effects.  :)

![Screenshot 2021-09-17 133332](https://user-images.githubusercontent.com/8879811/133844002-7e647009-3e57-47e3-a257-ea43e30841da.jpg)

Items 2 and 3 involve changes to editing an effect (on objects and layers).  There is currently just a short text above the input field that includes name + short description.  The long description should be moved below the input field to match function parameters and behavior properties ~and the top text should only include the parameter name (friendly + real).~ (removed, see below)

The "Show parameter names" checkbox can then be removed.

![Screenshot 2021-09-17 133122](https://user-images.githubusercontent.com/8879811/133844032-aec9750e-cc18-4aad-bc1d-162c7f948237.jpg)

*After thinking more, we NEVER need to show users the real parameter name if they always reference them using the pull-down lists.  This is similar to behavior properties and function parameters, users only see the friendly name.

**Comment by 4ian** [comment]
It's true that the completion in events would be more than welcome. I've made the changes to add this, see #3060.
